conjureup/ui/views/spellpicker.py

```diff
@@ -1,60 +1,100 @@
-from urwid import Filler, Pile, WidgetWrap
+from urwid import Filler, Pile, WidgetWrap, Frame, Text, Columns
 
 from ubuntui.ev import EventLoop
 from ubuntui.utils import Color, Padding
-from ubuntui.widgets.buttons import menu_btn, quit_btn
+from ubuntui.widgets.buttons import menu_btn
 from ubuntui.widgets.hr import HR
 
 
+class SpellPickerWidget(WidgetWrap):
+    def __init__(self, spell, cb):
+        self._spell = spell
+        self.cb = cb
+        super().__init__(self.build_widget())
+
+    def build_widget(self):
+        """ Provides a rendered spell widget suitable for pile
+        """
+        return Color.body(
+            menu_btn(label=self._spell['name'],
+                     on_press=self.cb,
+                     user_data=self._spell),
+            focus_map='menu_button focus'
+        )
+
+
 class SpellPickerView(WidgetWrap):
 
     def __init__(self, app, spells, cb):
         self.app = app
         self.cb = cb
-        self.spells = spells
+        self.spells = [SpellPickerWidget(spell, self.submit)
+                       for spell in spells]
         self.config = self.app.config
-        super().__init__(self._build_widget())
+        self.frame = Frame(body=self._build_widget(),
+                           footer=self._build_footer())
+        super().__init__(self.frame)
+        self.handle_focus_changed()
 
     def keypress(self, size, key):
+        rv = super().keypress(size, key)
         if key in ['tab', 'shift tab']:
             self._swap_focus()
-        return super().keypress(size, key)
+        self.handle_focus_changed()
+        return rv
+
+    def handle_focus_changed(self):
+        self.selected_spell_w = None
+        fw = self.pile.focus
+        if not isinstance(fw, SpellPickerWidget):
+            return
+
+        if fw != self.selected_spell_w:
+            self.selected_spell_w = fw
+            if fw is None:
+                self.spell_description.set_text("No spell selected")
+            else:
+                self.spell_description.set_text(fw._spell['description'])
 
     def _swap_focus(self):
-        w_count = len(self._w.body.contents) - 1
-        if self._w.body.focus_position >= 2 and \
-           self._w.body.focus_position < w_count:
-            self._w.body.focus_position = w_count
+        self.current_focused_frame = None
+        if self.current_focused_frame is None:
+            self.current_focused_frame = self.frame.footer
+            self.frame.footer.focus_position = 0
```

Comment:
@mikemccracken i was hoping just setting focus position to 0 in each of the frame's section would cause the first widget in that frame to be selected